### DISCUSSION TITLE
Unable to get MFA to appear after logging in

### DISCUSSION THREAD

**Discussion by Natakugithub** [neos-sdi/adfsmfa#325] [Q&A] [answered]
Hi there,

I have been struggling with getting it to work and trying to understand all the terms abit. I have mine setup with SQL, connection which looks ok and I read through all the documentation, but can't figure out what I'm missing. I would really appreciate some guidance on how to get it going.

The issue I'm having is everything seems enable, but the page never asks for the MFA code, it just logs in. We are using it on an webpage using ADFS to login. Not sure what other information is needed, please let me know if you need more information. I appreciate any help. Thanks,

Side note: The one other part I'm confused about is in the Security configuration, for Data Access Account SQL Server. What account am I suppose to put in there? I have the account adfs-mfa in AD as a test for the SQLServer configuration and it worked fine there, but if I put the same thing here the connection fails. A bit confused how these work.

**Comment by redhook62** [comment, marked as answer]
Hi, @Natakugithub 

If you have MFA configured just fine...
However you must tell ADFS to use MFA.
This is done in the ADFS console (and not the MFA) in the access policies and assign a rule which contains the activation of the MFA to your relying parties. 
In ADFS there is a default Access Policy that you can assign to your applications.

regards

**Reply by Natakugithub** [reply]
thanks for the update. I have tried looking at the main adfs console, but im not sure what I'm doing wrong. Also a side question, my sql connection string was working before but now suddenly throwing the error below.

CheckSQLConnection error : The value's length for key 'password' exceeds it's limit of '128'.

any idea what im missing?

-------------------------------------------------
this is what I have for the sql connection string (with some info replaced by generic things)

Integrated Security=SSPI;Persist Security Info=True;Initial Catalog=ADFSmfa2;Data Source=sqlsvrhostname;User ID=domainname\username;Password=Passwordchanged4secu!

-------------------------------------------------
here is some screenshot of my adfs
![Screenshot 2024-02-12 121017](https://github.com/neos-sdi/adfsmfa/assets/52261015/3bbf9b69-bf6a-4450-8ebc-95beeac9e74b)
![Screenshot 2024-02-12 121033](https://github.com/neos-sdi/adfsmfa/assets/52261015/9988e23f-3409-43c9-b9e0-0e6c45fed6bf)

**Reply by Natakugithub** [reply]
im so confused, the sql started working again on its own. does adfsmfa throw any errors on login if there is a problem? i can't tell if that popup on top right saying there is an error looking up for user is from the webapp the devs made or if its from adfsmfa

**Comment by redhook62** [comment]
Hi @Natakugithub 

Apparently you have correctly configured the MFA invocation in the ADFS console.

Then no, the MFA does not generate an error at the ADFS or SQLServer level. 

The SQL connection configured in the component is just used for storing and using each user's metadata in order to use MFA.

Check that your SQL connection is correct, but also check your environment which may be problematic.

regards

**Reply by Natakugithub** [reply]
ok thanks for the update, I will poke around and see what i can find. Thanks,
